ocw_import/api.py

```diff
@@ -74,13 +74,17 @@ def fetch_ocw2hugo_course_paths(bucket_name, prefix="", filter_list=None):
     s3 = get_s3_resource()
     bucket = s3.Bucket(bucket_name)
     paginator = bucket.meta.client.get_paginator("list_objects")
+    counter = 0
     for resp in paginator.paginate(Bucket=bucket.name, Prefix=f"{prefix}"):
```

Comment:
Instead of keeping a counter and returning when we find all our results, it might be faster to take a different approach.  This command is primarily used by `import_ocw_course_sites` and can only be run with a filter.  Therefore, it might be better here to instead iterate the `filter_list` and build the `Prefix` argument here using the courses in the list.  That would ensure that we are only querying S3 for the courses in the filter.  We might want to still check that `filter_list` is set before doing this as I believe `overwrite_ocw_course_content`, another management command for updating individual metadata properties, also uses this function and can still be run un-filtered.  Either way, I think if `filter_list` is set, we should only be querying S3 for the courses in that list.

Reply:
@gumaerc Yeah, I actually did not want to change the pagination mechanism and was a bit confused about the prefix as I thought different courses will have different  pefixes. Thats why I added this naive solution but your above explanation cleared everything and I have updated the code as per required.